### PR TITLE
[LOOM-614]: bpk accordion updates

### PR DESCRIPTION
**Pull request by mungodewar**
Change description: 
Hi folks 👋 

This is a change to the design of the accordion component to align with the design. I'll be including a screenshot of what has changed below. Namely: 

- Functionality to support the accordion on a dark background
- Divider behaviour change. The divider line between title & content has changed. There will no longer be a divider placed between title and content. This change will make more sense visually 👇  

Screenshot 2023-03-17 at 17 44 04
Screenshot 2023-03-17 at 17 44 10
Screenshot 2023-03-17 at 17 44 16

Remember to include the following changes:

- [ ] `README.md` (If you have created a new component)
- [x] Component `README.md`
- [x] Tests
- [x] Storybook examples created/updated

### CODE REVIEW

File: packages/bpk-component-accordion/src/BpkAccordionItem.module.scss [hunk 1]
```diff
@@ -24,7 +24,13 @@ $bpk-spacing-v2: true;
   &__title {
     height: auto;
 
-    @include bpk-border-bottom-sm($bpk-line-day);
+    &--collapsed {
+      @include bpk-border-bottom-sm($bpk-line-day);
+
+      &-on-dark {
+        @include bpk-border-bottom-sm($bpk-line-night);
```
Comment (mungodewar) [inline packages/bpk-component-accordion/src/BpkAccordionItem.module.scss:31]: Let's ask design about this (the night token usage)
Screenshot 2023-03-20 at 13 44 48

Comment (mungodewar) [reply packages/bpk-component-accordion/src/BpkAccordionItem.module.scss]: Updated the colours.

**Review by olliecurtis — CHANGES_REQUESTED** [review]
A small minor thing, but otherwise all looks great!

**Review by olliecurtis — APPROVED** [review]
LGTM!

Thank you gif

**Pull request merged**